meson.build

```diff
@@ -462,6 +462,41 @@ endif
 
 
 
+###############################################################
+# Library: Tcl (for pltcl)
+###############################################################
+
+tcl_dep = dependency('', required : false)
+tclopt = get_option('pltcl')
+if not tclopt.disabled()
+  possible_tcl_packages = ['tcl', 'tcl84', 'tcl8.4', 'tcl85', 'tcl8.5', 'tcl86', 'tcl8.6']
+
+  # via pkg-config
+  foreach t : possible_tcl_packages
+    tcl_dep = dependency(t, version : '>=8.4', required: tclopt)
+    if tcl_dep.found()
+      break
+    endif
+  endforeach
```

Comment:
Your current target meson version is `>=0.54`.

In 0.60.0 you can do this instead:
```meson
if not tclopt.disabled()
    possible_tcl_packages = ['tcl', 'tcl84', 'tcl8.4', 'tcl85', 'tcl8.5', 'tcl86', 'tcl8.6']
    tcl_dep = dependency(possible_tcl_packages, version : '>=8.4', required: false)
    if not tcl_dep.found()
        foreach t : possible_tcl_packages
            tcl_dep = cc.find_library(t,
            ...
```

Note that dependencies must be marked as `required: false` or else when tclopt is enabled, rather than auto, not finding it is immediately fatal even though you are going to check different names later on -- and fall back on cc.find_library too.

Speaking of which: is tcl likely to not have pkg-config support? Why?

Reply:
> Your current target meson version is >=0.54.

Due to the python version requirements there's no way we're going to require something as new as 0.60 unfortunately. This has been by *far* the biggest concern about adopting meson.

> Speaking of which: is tcl likely to not have pkg-config support? Why?

The only case I immediately found where there's no .pc file is system tcl on macos. Might also be similar on solaris / aix. So we might get away without the non-pkg-config path for tcl. For several of the other dependencies there's no .pc dependencies for more OSs (e.g. readline, openssl, ... on BSDs).

Reply:
Oh, looks like rhel / centos 7 also have no .pc file or tcl. Based on CI:
```
[07:51:23.786] Run-time dependency tcl found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl84 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl8.4 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl85 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl8.5 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl86 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Run-time dependency tcl8.6 found: NO (tried pkgconfig and cmake)
[07:51:23.786] Library tcl found: YES
[07:51:23.786] Has header "tcl.h" with dependency -ltcl: YES 
```
Unfortunately that means we'll need the fallback :(.

Looks like the .pc files were added in tcl 8.6, but rhel 7 still has 8.5.

Reply:
> Due to the python version requirements there's no way we're going to require something as new as 0.60 unfortunately. This has been by far the biggest concern about adopting meson.

Ah yeah, 0.54 through 0.56 all require python 3.5, but 0.57 bumped that to python 3.6 and 0.62 will bump it to python 3.7 again.

The OpenSSL one I implemented for 0.62.0 :) it is backwards compatible with 0.54.0, so you can simply tell people on BSD (?) that "it works if you use Meson 0.62.0"

If there is a strong need for this, I can consider adding a TCL dependency class too, and you can document that on macOS, tcl can only be found with Meson 0.63.0 (or whenever I land such a thing) and users of Meson 0.54.0 may not be able to enable tcl.

Reply:
(In order to implement a tcl dependency class, I'd need a good idea of how different versions with different names are supposed to be handled. Can this just be `'tcl'` handling either tcl.pc or a system class, with falling back to special versioned names? Why do these versioned names exist at all, do the platforms in question not also provide an unversioned tcl.pc?)

Reply:
Unfortunately only supporting ssl / tcl on newer meson versions (with higher python requirements) isn't going to fly, I'm afraid. So for now we're going to have to do it "by hand"...

> Why do these versioned names exist at all, do the platforms in question not also provide an unversioned tcl.pc?

Some platforms allow multiple versions to be installed in parallel, each with their own .pc. On freebsd at least there's no non-versioned .pc for tcl.